mongo/errors.go

```diff
@@ -202,13 +202,14 @@ type ServerError interface {
 	HasErrorMessage(string) bool
 	// HasErrorCodeWithMessage returns true if any of the contained errors have the specified code and message.
 	HasErrorCodeWithMessage(int, string) bool
-
-	serverError()
 }
 
 var _ ServerError = CommandError{}
 var _ ServerError = WriteException{}
 var _ ServerError = BulkWriteException{}
+var _ ServerError = driver.WriteCommandError{}
+var _ ServerError = driver.WriteConcernError{}
+var _ ServerError = driver.Error{}
```

Comment:
The issue that spurred this ticket was a deeply wrapped `driver.Error` that contained error labels: https://github.com/mongodb/mongo-go-driver/pull/651#issuecomment-827786843

Would a preferable alternative be to have the error translation functions translate wrapped errors as well? E.g. have `replaceErrors` also inspect wrapped errors?

@iwysiu would the fix of GODRIVER-1991 solve this problem entirely? If no errors are returned in `x`, I imagine the fix would involve having to rewrite wrapped errors as well.